extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,249 @@
+   * @param name
+   * @returns {object} client configuration
+   */
+  public getConfig(name?: string): object {
+    return name ? this.clients.find(client => client.name === name) : this.clients[0];
+  }
+
+  /**
+   * Gets the client list already setting the necessary wado headers
+   * @returns {Array} client list
+   */
+  public getClientsForWadoRequests() {
+    this.setWadoHeaders();
+    return this.getClients();
+  }
+
+  /**
+   * Gets the client list already setting the necessary qido headers
+   * @returns {Array} client list
+   */
+  public getClientsForQidoRequests() {
+    this.setQidoHeaders();
+    return this.getClients();
+  }
+
+  /**
+   * Returns the client list
+   * @returns {Array} client list
+   */
+  public getClients() {
+    return this.clients;
+  }
+  /**
+   * Adds a client to client list given a dicomweb server configuration
+   * @param configToAdd
+   * @returns {void}
+   */
+  private addClient(configToAdd: ConfigToAdd): void {
+    // if no qidoRoot or wadoRoot, don't add the client. Could be the case for
+    // configurations that relies on onConfiguration function but lacks necessary
+    // additional information
+    if (!configToAdd?.qidoRoot || !configToAdd?.wadoRoot) {
+      return;
+    }
+    const config = Object.assign({}, configToAdd);
+    config.qidoConfig = {
+      url: config.qidoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.wadoConfig = {
+      url: config.wadoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.qidoDicomWebClient = config.staticWado
+      ? new StaticWadoClient(config.qidoConfig)
+      : new api.DICOMwebClient(config.qidoConfig);
+
+    config.wadoDicomWebClient = config.staticWado
+      ? new StaticWadoClient(config.wadoConfig)
+      : new api.DICOMwebClient(config.wadoConfig);
+
+    config.qidoDicomWebClient.name = config.name;
+    config.wadoDicomWebClient.name = config.name;
+    this.clients.push(config);
+  }
+
+  /**
+   * Process a dicomweb server configuration and add it to the clients list.
+   * This function could change the configuration by calling the onConfiguration,
+   * if defined
+   * @param {object} params key / pair mapping of the URL parameters
+   * @param {object} query URLSearchParams object generated for the URL
+   * @param {object} config client configuration
+   * @returns {void}
+   */
+  private addConfiguration(params, query, config): void {
+    if (config.onConfiguration && typeof config.onConfiguration === 'function') {
+      config = config.onConfiguration(config, {
+        params,
+        query,
+      });
+    }
```

Comment:
you haven't added the client so i"m surprised this is running before the client creation

Reply:
If defined, onConfiguration is supposed to change the config before addition